packages/core/assistant/src/conversation/request.ts

```diff
@@ -0,0 +1,61 @@
+//
+// Copyright 2025 DXOS.org
+//
+
+import { type AiTool } from '@effect/ai';
+import { Cause, Effect, Exit, Fiber, type Layer } from 'effect';
+
+import { type AiModelNotAvailableError } from '@dxos/ai';
+import { runAndForwardErrors, throwCause } from '@dxos/effect';
+import { log } from '@dxos/log';
+import { type DataType } from '@dxos/schema';
+
+import { type AiSession, type AiSessionRunError, type AiSessionRunRequirements } from '../session';
+
+/**
+ * Request handle.
+ */
+export class AiConversationRequest<Tools extends AiTool.Any> {
+  private _fiber?: Fiber.Fiber<void, any>;
+
+  constructor(
+    private readonly _request: Effect.Effect<DataType.Message[], AiSessionRunError, AiSessionRunRequirements<Tools>>,
+    private readonly _session: AiSession,
+  ) {}
+
+  get session() {
+    return this._session;
+  }
+
+  /**
+   * Runs the request.
+   */
+  async run(services: Layer.Layer<AiSessionRunRequirements<Tools>, AiModelNotAvailableError>) {
```

Comment:
Any downside for using effect-based APIs? We'd have more consistency with conversation code